apps/frontend/src/models/interceptors/backend-url.interceptor.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpEvent, HttpHandler, HttpRequest, HttpResponse } from '@angular/common/http';
+import { TestBed } from '@angular/core/testing';
+import { APP_CONFIG } from 'common-frontend-models';
+import { API_BASE_SEGMENT } from 'common-shared-models';
+import { Observable, of } from 'rxjs';
+
+import { BackendUrlInterceptor } from './backend-url.interceptor';
+
+describe('BackendUrlInterceptor', () => {
+	let interceptor: BackendUrlInterceptor;
+	let mockHttpHandler: HttpHandler;
+	let mockHttpHandlerSpy: jest.SpyInstance;
+
+	beforeEach(() => {
+		mockHttpHandler = {
+			handle: (req: HttpRequest<object>): Observable<HttpEvent<object>> =>
+				of(new HttpResponse(req)),
+		} as HttpHandler;
+
+		TestBed.configureTestingModule({
+			providers: [
+				BackendUrlInterceptor,
+				{ provide: APP_CONFIG, useValue: { apiUrl: 'http://localhost:3000' } },
+				{ provide: API_BASE_SEGMENT, useValue: '/api' },
+			],
+		});
+
+		interceptor = TestBed.inject(BackendUrlInterceptor);
+		mockHttpHandlerSpy = jest.spyOn(mockHttpHandler, 'handle');
```

Comment:
Instead of defining the mock handler before each test case u could also define it once, and clear it beforeEach I think

Reply:
why should I do that? What are the benefits from it?

Reply:
Ok that is a good point, it wouldn't change anything